src/Fields.jl

```diff
@@ -47,3 +47,59 @@ function canonical_unit(x::FieldElement)
   iszero(x) && return one(x)
   return x
 end
+
+@doc raw"""
+   evaluation_points(K::Field, n::Int) -> Vector{FieldElem}
+   evaluation_points(K::AbstractAlgebra.Generic.RationalFunctionField, n::Int) -> Vector{RationalFunctionFieldElem}
+
+If $K$ contains at least $n$ elements, this function returns an $n$-element vector $v$ with pairwise different entries from the given field $K$. Otherwise it returns an empty vector. 
+"""
+function evaluation_points(K::Field, n::Int)
+   if n <= 0
+      error("n must be positive!")
+   end
+   if is_finite(K)
+      v = elem_type(K)[]
+      if (order(K) < n)
+         return v
+      else
+         for a in K
+            push!(v, a)
+            if length(v) == n
+               break
+            end
+         end
+      end
+   else
+      v = Vector{elem_type(K)}(undef, n)
+      v[1] = K(div(-n,2))
+      for i = 2:n
+         v[i] = v[i-1] + 1
+      end
+   end
+   return v
+end
+
+function evaluation_points(K::AbstractAlgebra.Generic.RationalFunctionField, n::Int)
+   if n <= 0
+      error("n must be positive!")
+   end
+   F = base_ring(K)
+   v = evaluation_points(F, n)
+   if is_empty(v)
+      v = elem_type(K)[]
+      base_v = evaluation_points(F, Int(order(F)))
+      d = ceil(Int, log(order(F), ZZ(n)))
+      for coeffs in AbstractAlgebra.ProductIterator(base_v, d; inplace=true)
+         push!(v, sum(coeffs[j] * gen(K)^(j-1) for j in 1:d))
```

Comment:
```suggestion
         push!(v, sum(coeffs[j] * genKpowers[j] for j in 1:d))
```
and then do `genKpowers = powers(gen(K), d - 1)` somewhere *outside* the loop.